protobuf/src/main/java/io/grpc/protobuf/StatusProto.java

```diff
@@ -103,6 +103,25 @@ public static StatusException toStatusException(
     return toStatus(statusProto).asException(toMetadata(statusProto, metadata));
   }
 
+  /**
+   * Convert a {@link com.google.rpc.Status} instance to a {@link StatusException} with additional
+   * metadata and the root exception thrown.
+   *
+   * <p>The returned {@link StatusException} will wrap a {@link Status} whose code and description
+   * are set from the code and message in {@code statusProto}. {@code statusProto} will be
+   * serialized and added to {@code metadata}. {@code metadata} will be set as the metadata of the
+   * returned {@link StatusException}. The {@link Throwable} is the exception that is set as the
+   * {@code cause} of the returned {@link StatusException}.
```

Comment:
Let's add a reminder here that cause is not transmitted over the wire.

Reply:
Added it in the first paragraph.

Reply:
What does "needs to be user provided" mean?